kore/src/SQL/SOP.hs

```diff
@@ -214,43 +234,45 @@ and tag columns are used as foreign keys into the constructor tables.
  -}
 createTableSum
     :: forall ctors
-    .  SOP.All2 Column ctors
+    .  SOP.All2 SOP.Top ctors
     => TableName
+    -> SOP.POP ColumnImpl ctors
     -> NP ConstructorInfo ctors
     -> SQL ()
-createTableSum tableName ctors = do
-    SOP.hctraverse_ proxyAllColumn (createConstructorTable tableName) ctors
+createTableSum tableName columnImpls ctors = do
+    SOP.hctraverse_
+        (Proxy @(SOP.All SOP.Top))
+        createTable'
+        (SOP.hzipWith Pair (SOP.unPOP columnImpls) ctors)
     createTable tableName names defs
   where
-    proxyAllColumn = Proxy @(SOP.All Column)
-
     names :: NP (K String) ctors
     names = columnNamesSum ctors
 
     defs :: NP (K ColumnDef) ctors
     defs = SOP.hmap (const $ K columnTag) names
 
-{- | @createTableGeneric@ implements 'createTable' for a 'SOP.Generic' type.
- -}
-createTableGeneric
-    :: forall proxy table
-    .  Typeable table
-    => (SOP.HasDatatypeInfo table, SOP.All2 Column (SOP.Code table))
-    => proxy table
-    -> SQL ()
-createTableGeneric proxy = createTableGenericAux (tableNameTypeable proxy) proxy
+    createTable'
+        :: forall fields
+        .  SOP.All SOP.Top fields
```

Comment:
I noticed there are multiple constraints such as this in the code. Are they still necessary? Doesn't it mean we aren't imposing any real constraints on the `fields`?

Reply:
Yes, they are necessary. This constraint means that `fields` is a type-level list of types `[Type]` with a known "shape", so that if we have a proxy type `Proxy fields`, we can construct any `fields`-based type, in particular `NP f fields` (the **N**-ary **P**roduct of `fields`).

Reply:
Right, I see, thanks!